### DISCUSSION TITLE
Allow to change payment method of an existing order to a hidden payment method

### DISCUSSION THREAD

**Discussion by ateuber** [pretix/pretix#3020] [Ideas]
A customer figured out that none of our payment methods works for them. Since bank transfer is more effort for us we show it only to people who enter the shop through a special link (`/unlock/hash/`). With this link only a new order seems to be possible, so the customer would need to enter all details again.

As a workaround we generate the invoice with the bank details, send it via email and once paid manually mark it as paid.

I think a better way would be to send a special link to the customer that allows to change to the hidden payment method for the existing order.

**Comment by raphaelm** [comment]
> With this link only a new order seems to be possible, so the customer would need to enter all details again.

that doesn't sound right, the link just sets a cookie that enables the payment method. so if they still have a cart running it can be completed. I'm not sure about unpaid orders, but if it's not respected there this could be fixed.

> I think a better way would be to send a special link to the customer that allows to change to the hidden payment method for the existing order.

How did they manage to create an order without selecting any of your payment methods?

**Reply by ateuber** [reply]
They selected giropay which then failed. When I open the unlock link in a separate browser and afterwards view the order as the customer and click on the button to try the payment again, it does not list the hidden method.

**Reply by raphaelm** [reply]
okay that might indeed be a bug
